spec/controllers/api/v2/application_controller_spec.rb

```diff
@@ -41,5 +42,33 @@
           .to eq('Token realm="Application"')
       end
     end
+
+    describe 'manage course api integration' do
+      let(:email) { "manage_courses@api.com" }
+      let(:sign_in_user_id) { "manage_courses_api" }
+      let(:existing_user) { create(:user, email: email, sign_in_user_id: sign_in_user_id) }
+      let(:secret) { Settings.authentication.secret = "SETTINGS:MANAGE_BACKEND:SECRET" }
+
+      before do
+        existing_user
+        Settings.authentication.secret = "SETTINGS:MANAGE_BACKEND:SECRET"
+      end
+
+      # NOTES:
+      # This is hardcode as the ruby version of JWT encoding is not symmetrical with csharp and does not applies
+      #
+      # RECOMMENDED (Notational Conventions) for the "typ" (Type) Header Parameter see (https://tools.ietf.org/html/rfc7519#section-5.1)
+      # as default.
+      # In any case as long as it can be verified it is fine.
+      let(:encoded_token) do
+        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzaWduX2luX3VzZXJfaWQiOiJtYW5hZ2VfY291cnNlc19hcGkiLCJlbWFpbCI6Im1hbmFnZV9jb3Vyc2VzQGFwaS5jb20ifQ.fs0tM5Lc_5vA2w94PhlDMnq50NUn2K5SMxggdAApp_w"
```

Comment:
How is this generated? It would be really nice to see that in the spec so we can understand it.

Reply:
> How is this generated?

https://github.com/DFE-Digital/manage-courses-api/pull/325/commits/6286fef6d2d55729301394bee07df77607ad1838

As explained 
`This is hardcode as the ruby version of JWT encoding is not symmetrical with csharp and does not applies...`

ie 
```
      JWT.encode(
        payload,
        Settings.authentication.secret,
        Settings.authentication.algorithm
      )
```

`JWT.encode` == `_manageCoursesBackendJwtService.GetCurrentUserToken()`

should of been equal except left hand side do `too little` and the right hand side do `too much`

configuring them to do the exact requires too much as the actual structure and ordering of items needs to exact, ie the header's items and the payload items

Reply:
Am I right to think that mcbe needs to send this across to mcapi (in the `Authorization` by the looks of it)? How does mcbe get it's hands on it? Is it in app's settings / environment?

Reply:
> Am I right to think that mcbe needs to send this across to mcapi (in the Authorization by the looks of it)?

No

Reply:
Its the otherway round...

mcapi will call mcbe 
mcapi http call Authorization header will contain the token in itself has generated

mcbe will authorise the call

Reply:
👍